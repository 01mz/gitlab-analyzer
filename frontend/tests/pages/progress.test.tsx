import React from 'react';

import Index from '../../pages/progress';
import {render} from "@testing-library/react";

describe("Pages Progress", () =>{
    const mockUseEffect = jest.spyOn(React, 'useEffect')

    it("Snapshot Index", () => {
        const { container } = render(
            <Index />
        )
        expect(container).toMatchSnapshot();
    })
    it("Test useEffect", ()=>{
        render(<Index />);
        expect(mockUseEffect).toBeCalled();
    })


})