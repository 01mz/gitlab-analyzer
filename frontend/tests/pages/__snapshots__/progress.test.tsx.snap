// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages Progress Snapshot Index 1`] = `
<div>
  <div
    class="MuiBox-root MuiBox-root-3 makeStyles-backgroundGradient-1"
  >
    <div
      class="MuiBox-root MuiBox-root-4"
    >
      <div
        class="MuiBox-root MuiBox-root-5"
        id="card-header"
      >
        <div
          style="display: inline-block; max-width: 100%; overflow: hidden; position: relative; box-sizing: border-box; margin: 0px;"
        >
          <div
            style="box-sizing: border-box; display: block; max-width: 100%;"
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;base64,PHN2ZyB3aWR0aD0iMTI1IiBoZWlnaHQ9IjEyNSIgeG1sbnM9Imh0dHA6Ly93d3cudzMub3JnLzIwMDAvc3ZnIiB2ZXJzaW9uPSIxLjEiLz4="
              style="max-width: 100%; display: block; margin: 0px; padding: 0px;"
            />
          </div>
          <img
            alt="The Gitlab Logo"
            decoding="async"
            src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
            style="visibility: hidden; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
          />
        </div>
        <h6
          class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom"
        >
          GitLab
          <br />
          Analyzer
        </h6>
      </div>
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="MuiLinearProgress-root WithStyles(ForwardRef(LinearProgress))-root-6 MuiLinearProgress-colorPrimary WithStyles(ForwardRef(LinearProgress))-colorPrimary-7 MuiLinearProgress-determinate"
        role="progressbar"
      >
        <div
          class="MuiLinearProgress-bar WithStyles(ForwardRef(LinearProgress))-bar-8 MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate"
          style="transform: translateX(-100%);"
        />
      </div>
      <div
        class="MuiBox-root MuiBox-root-9"
      >
        <p
          class="makeStyles-progress_dots-11 false"
        >
          Importing commits
        </p>
        <p
          class="makeStyles-progress_dots-11 false"
        >
          Importing merge requests
        </p>
        <p
          class="makeStyles-progress_dots-11 false"
        >
          Importing comments
        </p>
        
      </div>
      <div
        class="MuiBox-root MuiBox-root-13"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-14 MuiButton-containedPrimary MuiButton-disableElevation"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             Cancel 
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</div>
`;
